collector/exporter.go

```diff
@@ -104,37 +98,52 @@ func New(ctx context.Context, dsn string, metrics Metrics, scrapers []Scraper, l
 		logger:   logger,
 		dsn:      dsn,
 		scrapers: scrapers,
-		metrics:  metrics,
+		up: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "up"),
+			"Whether the MySQL server is up.",
+			nil,
+			nil,
+		),
+		scrapeFailed: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, exporter, "last_scrape_failed"),
+			"Whether the last scrape of metrics from MySQL resulted in an error (1 for error, 0 for success).",
+			nil,
+			nil,
+		),
+		scrapeCollectorSuccess: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, exporter, "collector_success"),
+			"mysqld_exporter: Whether a collector succeeded.",
+			[]string{"collector"},
+			nil,
+		),
+		scrapeDurationSeconds: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, exporter, "collector_duration_seconds"),
+			"Collector time duration.",
+			[]string{"collector"}, nil,
+		),
 	}
 }
 
 // Describe implements prometheus.Collector.
 func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
-	ch <- e.metrics.TotalScrapes.Desc()
-	ch <- e.metrics.Error.Desc()
-	e.metrics.ScrapeErrors.Describe(ch)
-	ch <- e.metrics.MySQLUp.Desc()
+	ch <- e.up
```

Comment:
For example here, if there was a `mysqlUpDesc` package var, you could do this:

```suggestion
	ch <- mysqlUpDesc
```